actions/knowledge/knowledge.ts

```diff
@@ -21,3 +23,118 @@ export async function ingest(
   );
   return;
 }
+
+export async function deleteDataset(datasetID: string): Promise<void> {
+  const datasetDir = path.join(KNOWLEDGE_DIR(), 'script_data', datasetID);
+  if (!fs.existsSync(datasetDir)) {
+    return;
+  }
+  await execPromise(`${process.env.KNOWLEDGE_BIN} delete-dataset ${datasetID}`);
+  await fs.promises.rm(datasetDir, {
+    recursive: true,
+    force: true,
+  });
+  return;
+}
+
+export async function ensureFilesIngested(
+  files: string[],
+  scriptId: string,
+  token: string
+) {
+  const dir = path.join(KNOWLEDGE_DIR(), 'script_data', scriptId, 'data');
+  if (!fs.existsSync(dir) && files.length > 0) {
+    fs.mkdirSync(dir, { recursive: true });
+  } else if (!fs.existsSync(dir) && files.length === 0) {
+    // if there are no files in the directory and no dropped files, do nothing
+    return;
+  }
+
+  for (const file of files) {
+    const filePath = path.join(dir, path.basename(file));
+    try {
+      if (!fs.existsSync(filePath)) {
+        await fs.promises.copyFile(file, filePath);
+      }
+    } catch (error) {
+      console.error(`Error copying file ${file}:`, error);
+    }
+  }
+
+  try {
+    const filesInDir = await fs.promises.readdir(dir);
+    for (const fileName of filesInDir) {
+      const fullPath = path.join(dir, fileName);
+      const fileInDroppedFiles = files.find(
+        (file) => path.basename(file) === path.basename(fullPath)
+      );
+      if (!fileInDroppedFiles || !files || files.length === 0) {
+        await fs.promises.unlink(fullPath);
+      }
+    }
+  } catch (error) {
+    console.error('Error during cleanup of removed files:', error);
+  }
+
+  try {
+    await runKnowledgeIngest(
+      scriptId,
+      path.join(KNOWLEDGE_DIR(), 'script_data', scriptId),
+      token
+    );
+  } catch (error) {
+    console.error('Error during ingestion:', error);
```

Comment:
We discussed via Slack that we could as well just throw the error and let the framework handle that